include/deal.II/sundials/n_vector.templates.h

```diff
@@ -528,41 +598,20 @@ namespace SUNDIALS
 
 
 
-      template <typename VectorType,
-                std::enable_if_t<IsBlockVector<VectorType>::value, int>>
-      const MPI_Comm &
-      get_communicator(N_Vector v)
-      {
-        return unwrap_nvector_const<VectorType>(v)
-          ->block(0)
-          .get_mpi_communicator();
-      }
-
-
-
-      template <typename VectorType,
-                std::enable_if_t<!IsBlockVector<VectorType>::value, int>>
+      template <typename VectorType>
       const MPI_Comm &
       get_communicator(N_Vector v)
       {
-        return unwrap_nvector_const<VectorType>(v)->get_mpi_communicator();
-      }
-
-
-
-      template <typename VectorType,
-                std::enable_if_t<is_serial_vector<VectorType>::value, int>>
-      void *get_communicator_as_void_ptr(N_Vector)
-      {
-        // required by SUNDIALS: MPI-unaware vectors should return the nullptr
-        // as comm
-        return nullptr;
+        Assert(v != nullptr, ExcInternalError());
+        Assert(v->content != nullptr, ExcInternalError());
+        auto *pContent =
+          reinterpret_cast<NVectorContent<VectorType> *>(v->content);
+        return pContent->get_mpi_communicator();
       }
```

Comment:
Why did this disappear? Is `&MPI_COMM_SELF==nullptr`?

Reply:
There were two specializations of the function, one for serial and one for parallel vectors (enabled or disabled via `std::enable_if`) because the former do not have a `get_communicator()` member function. 

I've outsourced that decision to the `get_mpi_communicator()` free function above now that returns `MPI_COMM_SELF` for serial vectors. Are you saying that we should just return `nullptr` if a vector is serial, regardless of whether we configured for MPI or not?

Reply:
Maybe something like this in the current place?
```
diff --git a/include/deal.II/sundials/n_vector.templates.h b/include/deal.II/sundials/n_vector.templates.h
index e5e9db4bd2..e4a42d659a 100644
--- a/include/deal.II/sundials/n_vector.templates.h
+++ b/include/deal.II/sundials/n_vector.templates.h
@@ -619,8 +619,12 @@ namespace SUNDIALS
         (void)v;
         return nullptr;
 #  else
-        // We need to cast away const here, as SUNDIALS demands a pure `void *`.
-        return &(const_cast<MPI_Comm &>(get_communicator<VectorType>(v)));
+        if (is_serial_vector<VectorType>::value)
+          // We need to cast away const here, as SUNDIALS demands a pure `void
+          // *`.
+          return &(const_cast<MPI_Comm &>(get_communicator<VectorType>(v)));
+        else
+          return nullptr;
 #  endif
       }
 
```

Reply:
I convinced myself that that is the right way to go and pushed an updated commit.